### ISSUE TITLE
Implement gradient clipping to optimizers

### ISSUE THREAD

**Issue by jessegrabowski** [jessegrabowski/pytensor-ml#5]
Add some way to clip gradients, either to a specified norm or via elementwise clip. Right now it would just be a helper inside the training loop, because we are just appending the optimizer "parameters" (i.e. the gradients of the weights) to the list parameters themselves and passing that recursively into an update function. I did it this way because I wanted to avoid using shared variables and updates. I recognize, however, that this desire to avoid shared variables is totally irrational, and should probably be revisited for issues like this one and #6
